app/packages/looker/src/worker/painter.ts

```diff
@@ -224,12 +234,20 @@ export const PainterFactory = (requestColor) => ({
       maskData.buffer = maskData.buffer.slice(0, overlay.length);
     } else {
       const cache = {};
-
       let color;
-      if (maskTargets && Object.keys(maskTargets).length === 1) {
+      let colorKey;
+      if (maskTargets && Object.keys(maskTargets).length === 0) {
+        if (coloring.by === "field") {
+          colorKey = field;
+        } else {
+          colorKey = label.id;
+        }
+
         color = get32BitColor(
-          setting?.fieldColor ??
-            (await requestColor(coloring.pool, coloring.seed, field))
+          coloring.by === "field"
+            ? setting?.fieldColor ??
+                (await requestColor(coloring.pool, coloring.seed, colorKey))
+            : await requestColor(coloring.pool, coloring.seed, colorKey)
```

Comment:
nit: no need to have two calls to `requestColor` with the same arguments
```suggestion
        const requestedColor =
          coloring.by === "field" && setting?.fieldColor
            ? setting?.fieldColor
            : await requestColor(coloring.pool, coloring.seed, colorKey);
        color = get32BitColor(requestedColor);
```